docs/integrations/webhooks/debugging-webhooks.mdx

```diff
@@ -0,0 +1,93 @@
+
+Below is an image demonstrating the flow from when you perform an action that triggers a webhook through until the webhook is received by your application
+
+FLOW CHART IMAGE HERE
+
+
+## Debugging
+
+With the Webhook flow in mind, the following areas could cause problems when trying to develop with webhooks:
+
+1. The webhook route is not correctly configured
+2. Configuration in your dashboard
+3. The tunnel being used
+4. Verification of the webhook
+5. The logic you created to process the webhook
+
+
+### Testing your Webhook route locally
+
+If you are having trouble with your webhook, you can create a basic response and then test it locally. First, add the following file and code to your app. Also make sure that you add `ignoredRoutes: ['/api/webhooks(.*)']` to your `authMiddleware()`.
+
+```ts filename="app/webhooks/test/route.ts"
+export async function POST() {
+   return Response.json({ message: 'The webhook route is working' });
+}
+```
+
+While your application is running, try this from the command line.
+
+```bash
+curl -H 'Content-Type: application/json' \
+     -X POST http://localhost:3000/api/webhooks/test
+```
+
+If you see the `{"message":"The webhook route is working"}` then the basic webhook route is working, and ready to build on.
+
+
+### Configuration in your dashboard
+
+Whether you are developing locally or deploying to production, the webhook URL provided must be exact and correct. The URL breaks down into three parts -- the protocol (`http` vs `https`), the domain (`domain.com`) and the path (`/api/webhooks/user`).
+- Whether in development using a tunnel or in production, the URL will almost always use `https`.
+- The domain needs to be exact. A common error in production is not including the `www.`. Unlike entering a domain in your browser, a webhook will not be redirect from `domain.com` to `www.domain.com`. If your application lives on `www.domain.com` then the webhook URL must use that.
+- The path must match the path in your application.
+
+### Debugging your tunnel
+
+You've setup your route and tested it, configured your tunnel, updated your dashboard and now are trying to use the webhook but are getting errors. You will want to visit the [Webhook section](https://dashboard.clerk.com/last-active?path=webhooks) of your dashboard and the select the webhook you created.
+
+You will see a 'Webhook Attempts' section and likely see that one or more of those attempts have failed. Click on the failed attempt to expand it. In the details for the attempt, there will be a 'HTTP RESPONSE CODE'. The code will likely be either a 500 or something starting with a 4, like 401 or 404. See [HTTP response status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) from MDN for reference.
+
+IMAGES FROM THE WEBHOOK SECTION HERE
+
+Below is a table with some of the common response codes you might see, and what they indicate:
```

Comment:
I think it's worth giving this its own H3 heading. "Webhook error response codes" for example. I can see someone wanting to link to this section, or jump straight to it from the TOC

Reply:
Added an H3, but used `### HTTP Response Codes` since these are not specific to webhooks.